lib/administrate/field/number.rb

```diff
@@ -30,6 +33,11 @@ def decimals
       def value
         data * options.fetch(:multiplier, 1)
       end
+
+      def add_delimiter(result)
+        ActiveSupport::NumberHelper.
+          number_to_delimited(result, delimiter: options[:delimiter])
```

Comment:
I'm not sure about the user passing a specific delimiter explicitly. In theory, this should be provided by the locale. In fact `number_to_delimited` should by the default format as per the locale, so it shouldn't need a delimiter either.

So in my mind, `:delimiter` should be just a boolean option. Thoughts?

Reply:
It would be great to have a way to give various options that then get forwarded to `number_to_delimited`: 
- delimiter
- separator
- delimiter_pattern

However the same could be said for other ActiveSupport methods, e.g `number_to_percentage`:
- precision
- significant
- separator
- delimiter
- strip_insignificant_zeros
- format

+ `number_to_rounded`, `number_to_phone`, `number_to_currency`, `number_to_human_size`, `number_to_human`. So even if all of that is not enabled in this PR it might be good to have an API that does not prevent these other possibilities.

For example this field could accept two options:
- `conversion_type` with possible values `number_to_delimited`, `number_to_percentage`, etc
- `conversion_options` would be passed transparently to the relevant ActiveSupport method

Reply:
Thanks for the comments, @pablobm & @sedubois. 

>  In theory, this should be provided by the locale. In fact `number_to_delimited` should by the default format as per the locale, so it shouldn't need a delimiter either. So in my mind, `:delimiter` should be just a boolean option. Thoughts?

Personally, while I imagine that in the majority of cases users will default to whatever is standard for their locale, when using Administrate I would appreciate the added flexibility of being able to choose my own delimiter. 

> It would be great to have a way to give various options that then get forwarded to `number_to_delimited`

I like this idea, and the idea of leaving the API "ready" for future updates to allow other ActiveSupport methods. On note, however, is that as there's already a `decimals` option, if we were to do so, we'd need to make sure that `separator` from `number_to_delimited` wouldn't step on its toes. 

It looks like there are three options.

1. Do as @pablobm suggests and convert `:delimiter` to just being a boolean.
2. Keep as is (i.e., explicit defining of delimiter), perhaps adding the option to pass `true` to default to the delimiter for the user's locale.
3. Update to allow passing of various options to `number_to_delimited`, as @sedubois suggests.

Thoughts? I'm happy to implement any of these the maintainers/etc. agree is the preferred direction they/you want to take Administrate.

Reply:
Good point on other available number helpers, @sedubois! This does complicate things. Still, I think it's worth looking into this issue, as `Number` is already a bit locale-unfriendly with its use of a format string like `"%.#{decimals}f"`.

Your idea of having two arguments `conversion_type` and `conversion_options` sounds appealing to me. One step further would be to allow a lambda to do the conversion, but I feel that could go too far, and `Number` could become a "universal" type of field.

Perhaps use different names, like `formatter` and `formatter_options`? I think this describes their intent better.

Reply:
> Perhaps use different names, like formatter and formatter_options

LGTM 👍

Reply:
Thanks, @pablobm. I've now implemented this. Please take a look when you get the chance.